src/dodal/testing/fixtures/run_engine.py

```diff
@@ -3,32 +3,70 @@
 """
 
 import asyncio
+import threading
 import time
 from collections.abc import Mapping
 
 import pytest
+import pytest_asyncio
 from bluesky.run_engine import RunEngine
 from bluesky.simulators import RunEngineSimulator
 
-_run_engine = RunEngine()
-
 
 @pytest.fixture(scope="session", autouse=True)
-async def _ensure_running_bluesky_event_loop():
+async def _ensure_running_bluesky_event_loop(_global_run_engine):
     # make sure the event loop is thoroughly up and running before we try to create
     # any ophyd_async devices which might need it
     timeout = time.monotonic() + 1
-    while not _run_engine.loop.is_running():
+    while not _global_run_engine.loop.is_running():
         await asyncio.sleep(0)
         if time.monotonic() > timeout:
             raise TimeoutError("This really shouldn't happen but just in case...")
 
 
 @pytest.fixture()
-def run_engine():
-    global _run_engine
-    _run_engine.reset()
-    return _run_engine
+async def run_engine(_global_run_engine: RunEngine):
+    try:
+        yield _global_run_engine
+    finally:
+        _global_run_engine.reset()
+
+
+@pytest_asyncio.fixture(scope="session", loop_scope="session")
+async def _global_run_engine():
+    """
+    Obtain a run engine, with its own event loop and thread.
+
+    On closure of the scope, the run engine is stopped and the event loop closed
+    in order to release all resources it consumes.
+    """
+    run_engine = RunEngine({}, call_returns_result=True)
+    yield run_engine
+    try:
+        run_engine.halt()
+    except Exception as e:
+        print(f"Got exception while halting RunEngine {e}")
```

Comment:
I personally think just do 

```suggestion
    run_engine.halt()
```
If there's an error halting, we shouldn't just print it, we should let it fail naturally and let test crash so very obvious something has gone wrong